PBT/test/Test/Compiler.hs

```diff
@@ -0,0 +1,78 @@
+module Test.Compiler (props) where
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import HW.Compiler
+import HW.StackMachine
+import HW.Eval
+
+import qualified Expr as E
+
+import Test.Tasty
+import Test.Tasty.Hedgehog
+
+-- Generator for numeric expressions
+genNumExpr :: Gen (E.Expr String)
+genNumExpr = E.Num <$> Gen.int (Range.linear (-1000) 1000)
+
+-- Generator for addition expressions
+genAddExpr :: Gen (E.Expr String)
+genAddExpr = do
+  left <- genNumExpr
+  right <- genNumExpr
+  return $ E.Plus left right
```

Comment:
These are rather weak generators. Use `Gen.recursive` to generate random expressions and test them.